### PR TITLE
Copy tested AMI to more regions

### PR DESCRIPTION
**Pull request by cgwalters**
Closes: https://github.com/openshift/os/issues/328

### CODE REVIEW

**Comment by cgwalters** [comment]
Still playing with/testing this - it looks like we at least also need to replicate the launch permissions, and possibly the tags.

**Comment by cgwalters** [comment]
OK so I updated this to support copying tags and making the image public.  Currently hitting:

```
$ env AWS_PROFILE=aws-ci ~/src/github/openshift/os/scripts/ami-copy-regions --source-region us-east-1 --source-image-id ami-01be9707253b2d46d --regions us-east-2,us-west-1 --copy-tags rhcos_tag,ostree_version --name rhcos_dev_WALTERS --out amis.json
Tags to copy: ['Key=rhcos_tag,Value=alpha', 'Key=ostree_version,Value=4.0.6225']
Uploading to: us-east-2
Complete, ImageId=ami-0a0d75593867c33a7
Copying tags...
Making public for launch

An error occurred (InvalidAMIID.Unavailable) when calling the ModifyImageAttribute operation: The AMI ID 'ami-0a0d75593867c33a7' is currently pending and may not be used for this operation
```

AFAICS, mantle has some code to retry/poll for GCE, but not for AWS?  Googling for the error I see other people hitting it though, looks like at least...

Oh wait, hey there's https://docs.ansible.com/ansible/devel/modules/ec2_ami_copy_module.html#ec2-ami-copy-module

**Comment by cgwalters** [comment]
Some more investigation turns up the handy `aws ec2 wait image-available` incantation which is obviously what we want.  Updated :arrow_up:  (But still testing locally, whee I'm leaking AMIs...)

**Comment by cgwalters** [comment]
OK, the script seems to work for me locally.   I didn't test the `Jenkinsfile` changes.

Lifting WIP - but let's wait to merge till next Monday at the earliest.

**Comment by cgwalters** [comment]
One thing I want to note about this is that the "AMI JSON" format intentionally matches the existing Container Linux schema.

File: Jenkinsfile.aws-test [hunk 1]
```diff
@@ -66,11 +71,19 @@ node(NODE) {
...
+                           --out ${WORKSPACE}/aws-tested.json
 
-                    # Upload the json file to a public location
-                    aws s3 cp --acl public-read \
-                        ${WORKSPACE}/aws-${AWS_REGION}.json \
-                        s3://${S3_PUBLIC_BUCKET}/aws-${AWS_REGION}-tested.json
+                    # Upload the json files to a public location
+                    for k in aws-${AWS_REGION}-tested.json aws-tested.json; do
+                        aws s3 cp --acl public-read ${WORKSPACE}/\${k} \
+                            s3://${S3_PUBLIC_BUCKET}/\${k}
```
[the remark this answers is not in the document]
Comment (yuqi-zhang) [reply Jenkinsfile.aws-test]: As in remove `aws-us-test-1-tested.json` and have a list in `aws-tested.json`? I think that makes sense.

File: Jenkinsfile.aws-test [hunk 2]
```diff
@@ -66,11 +71,19 @@ node(NODE) {
...
+                           --copy-tags rhcos_tag,ostree_version,ostree_commit \
+                           --regions ${OTHER_AWS_REGIONS.join(',')} \
+                           --out ${WORKSPACE}/aws-tested.json
 
-                    # Upload the json file to a public location
-                    aws s3 cp --acl public-read \
-                        ${WORKSPACE}/aws-${AWS_REGION}.json \
-                        s3://${S3_PUBLIC_BUCKET}/aws-${AWS_REGION}-tested.json
+                    # Upload the json files to a public location
+                    for k in aws-${AWS_REGION}-tested.json aws-tested.json; do
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply Jenkinsfile.aws-test]: OK I rebased and fixed this  :arrow_down: , thanks for spotting it.

**Pull request merged**